components/file_limit_sticky_banner/index.tsx

```diff
@@ -0,0 +1,185 @@
+border: none;
+outline: none;
+background: none;
+position: absolute;
+top: 16px;
+right: 0;
+`;
+
+const StyledI = styled.i`
+color: #FFBC1F;
+`;
+
+function FileLimitStickyBanner() {
+    const [show, setShow] = useState(true);
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch();
+    const [cloudLimits, limitsLoaded] = useGetLimits();
+    const openPricingModal = useOpenPricingModal();
+
+    const user = useSelector(getCurrentUser);
+    const isAdmin = useSelector(isCurrentUserSystemAdmin);
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const product = useSelector(selectSubscriptionProduct);
+    const isStarter = product?.sku === CloudProducts.STARTER;
+
+    const snoozePreferenceVal = useSelector((state: GlobalState) => selectPreference(state, Preferences.CLOUD_USER_EPHEMERAL_INFO, 'file_limit_banner_snooze'));
+
+    let shouldShowAgain = true;
+    if (snoozePreferenceVal !== '') {
+        const snoozeInfo = JSON.parse(snoozePreferenceVal);
+        const snoozeCoolOffDaysMillis = snoozeCoolOffDays * 24 * 60 * 60 * 1000;
+        const timeDiff = Date.now() - snoozeInfo.LastSnoozeTimestamp;
+        shouldShowAgain = timeDiff >= snoozeCoolOffDaysMillis;
+    }
+
+    if (!show) {
+        return null;
+    }
+
+    if (!shouldShowAgain) {
+        return null;
+    }
+
+    if (!isCloud || !isStarter) {
+        return null;
+    }
+
+    let isAbovePlanFileStorageLimit = false;
+
+    if ((limitsLoaded || hasSomeLimits(cloudLimits))) {
+        const currentFileLimitGB = cloudLimits.files?.total_storage || fallbackStarterLimits.files.totalStorage;
+        if (currentFileLimitGB > maxStarterPlanFileStorageGB) {
+            isAbovePlanFileStorageLimit = true;
+        }
+    }
+
+    if (isAbovePlanFileStorageLimit) {
+        return null;
+    }
+
+    const snoozeBanner = () => {
+        const fileLimitBannerSnoozeInfo = {
+            LastSnoozeTimestamp: Date.now(),
+        };
+
+        dispatch(savePreferences(user.id, [
+            {
+                category: Preferences.CLOUD_USER_EPHEMERAL_INFO,
+                name: 'file_limit_banner_snooze',
+                user_id: user.id,
+                value: JSON.stringify(fileLimitBannerSnoozeInfo),
+            },
+        ]));
+
+        setShow(false);
+    };
+
+    const AdminMessageLink = (
+        <a
+            onClick={(e) => {
+                e.preventDefault();
+                openPricingModal();
+            }}
+        >
+            {formatMessage({id: 'create_post.file_limit_sticky_banner.admin_link_message', defaultMessage: 'upgrade to a paid plan.'})}
+        </a>
+    );
+
+    const AdminMessage = (
+        <span>
+            {formatMessage({
+                id: 'create_post.file_limit_sticky_banner.admin_message',
+                defaultMessage: 'Your free plan is limited to {storageGB}GB of files. New uploads will automatically archive older files. To view them again, you can delete older files or '},
```

Comment:
Can we combine this into a single message since it is a single logical message to the user? Some advantages are:
* It will be easier for translators to understand the context of the link message.
* It avoids scenarios where translation happened for one string but not the other, and so now for some languages half the message is translated but the other half is not and defaults to English.

```tsx
            {formatMessage({
                id: 'create_post.file_limit_sticky_banner.admin_message',
                defaultMessage: 'Your free plan is limited to {storageGB}GB of files. New uploads will automatically archive older files. To view them again, you can delete older files or <a>upgrade to a paid plan.</a>'},
            {
                storageGB: fallbackStarterLimits.files.totalStorage / FileSizes.Gigabyte,
                a: (chunks: React.ReactNode) => {
                  return (
                    <a
                     onClick={
                      (e) => {
                        e.preventDefault();
                        openPricingModal();
                      }
                   }>
                    {chunks}
                   </a>
                );
            },
            )}
```

Reply:
Thanks. I didn't know I could format like this.